src/omni_epd/displays/inky_display.py

```diff
@@ -48,111 +52,71 @@ def __init__(self, deviceName, config):
             self._device = deviceObj.InkyWHAT(dColor)
+        elif(dType == 'impression'):
+            deviceObj = self.load_display_driver(self.pkg_name, 'inky_uc8159')
+            self._device = deviceObj.Inky(dColor)
+            self.mode = dColor = 'color' # default to color for impression
         elif(dType == 'auto'):
             deviceObj = self.load_display_driver(self.pkg_name, 'auto')
             self._device = deviceObj.auto()
             dColor = 'color' if self._device.colour == 'multi' else self._device.colour
 
+        self.clear_color = deviceObj.WHITE
+
         # set mode to black + any other color supported
         if(self.mode != "black"):
             self.modes_available = ('black', dColor)
 
-        # phat and what devices expect colors in the order white, black, other
+        # phat and what devices expect colors in the order white, black, other, 
         if(self.mode == "red" and dColor == "red"):
             self.palette_filter.append([255, 0, 0])
             self.max_colors = 3
         elif(self.mode == "yellow" and dColor == "yellow"):
             self.palette_filter.append([255, 255, 0])
             self.max_colors = 3
+        elif(self.mode == "color" and dColor == "color"):
+            self.palette_filter = deviceObj.DESATURATED_PALETTE
+            self.max_colors = 8
+            self.clear_color = deviceObj.CLEAN
 
         # set the width and height
         self.width = self._device.width
         self.height = self._device.height
 
     @staticmethod
     def get_supported_devices():
-        result = []
-
-        deviceList = ["phat_black", "phat_red", "phat_yellow",
-                      "phat1608_black", "phat1608_red", "phat1608_yellow",
-                      "what_black", "what_red", "what_yellow", "auto"]
-
-        # python libs for this might not be installed - that's ok, return nothing
-        if(check_module_installed(INKY_PKG)):
-            # if passed return list of devices
-            result = [f"{INKY_PKG}.{n}" for n in deviceList]
-
-        return result
+        return [] if not check_module_installed(INKY_PKG) else [f"{INKY_PKG}.{n}" for n in InkyDisplay.deviceList]
 
+    # set the image and display
     def _display(self, image):
+        # set border
+        self._device.set_border(getattr(self._device, self._get_device_option('border', '').upper(), self._device.border_colour))
+        
         # apply any needed conversions to this image based on the mode
-        image = self._filterImage(image)
+        if(self.mode == "color"):
+            # dont filter the image if it's an impression, only set saturation
+            saturation = self._getfloat_device_option('saturation', .5)  # .5 is default from Inky lib
+            self._device.set_image(image.convert("RGB"), saturation=saturation) 
+        else:
+            image = self._filterImage(image)
+            self._device.set_image(image)

```

Comment:
```suggestion
            self._device.set_image(image.convert("RGB"))
```
Need to convert to RGB, otherwise black mode is inverted on impression. Should be okay for other displays as well.